### PR TITLE
Add option to change the distance matrix computation

### PR DESCRIPTION
**Pull request by lbugnon**
This is a small change to use the average distance instead of summed distance of neighbours. This is to avoid  border neurons having different distance value computation. 
The original behaviour is set by default.

### CODE REVIEW

**Comment by JustGlowing** [comment]
Thank you, I'll have a look within a week.

File: minisom.py [hunk 1]
```diff
@@ -434,14 +435,19 @@ def train_batch(self, data, num_iteration, verbose=False):
         """
         self.train(data, num_iteration, random_order=False, verbose=verbose)
 
-    def distance_map(self):
+    def distance_map(self, neighbour_average=False):
```
Comment (JustGlowing) [inline minisom.py:438]: I would call the new parameter `scaling` and have it as a string. This way the user can scale by the sum using the method like this

```
distance_map(scaling='sum')
```

or scale by the average using it like this

```
distance_map(scaling='mean')
```

File: tests/test_distance_map_average.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+from minisom import MiniSom
+from sklearn.datasets import load_iris
+from matplotlib import pyplot as plt
+
+
+def test_distance_map():
+    """
+    Compare distance maps with averaged or added neighbour distances.
```
Comment (JustGlowing) [inline tests/test_distance_map_average.py:8]: all the tests of MiniSom are in `minisom.py` and there's already a test for `distance_map`: https://github.com/JustGlowing/minisom/blob/master/minisom.py#L753

File: tests/test_distance_map_average.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    dist_map = som.distance_map()
+    # map with averaged distances
+    dist_map_average = som.distance_map(neighbour_average=True)
+
+    fig, ax = plt.subplots(1, 2)
+    ax[0].imshow(dist_map)
+    ax[0].set_title("Default map")
+    ax[1].imshow(dist_map_average)
+    ax[1].set_title("Averaged map")
+    plt.show()
```
Comment (JustGlowing) [inline tests/test_distance_map_average.py:24]: We can't show a chart every time unit tests are run.

**Comment by lbugnon** [comment]
@JustGlowing addressed the comments

File: minisom.py [hunk 4]
```diff
@@ -435,7 +435,7 @@ def train_batch(self, data, num_iteration, verbose=False):
         """
         self.train(data, num_iteration, random_order=False, verbose=verbose)
 
-    def distance_map(self, neighbour_average=False):
+    def distance_map(self, scaling='sum'):
```
Comment (JustGlowing) [inline minisom.py:438]: changes look great. Can you fix the documentation of the method and a check to validate scaling?

Here's an example of how parameters are checked in other parts of the library https://github.com/JustGlowing/minisom/blob/master/minisom.py#L169

After this I'll run the tests myself and, if they pass, I'll merge the branch.

**Comment by lbugnon** [comment]
@JustGlowing sorry always forget about documentation and validation. I think it's good now

**Comment by JustGlowing** [comment]
@lbugnon  Thanks for you amazing contribution. I'm going to merge this now!

**Pull request merged**